File: copy/pars/parsing/models/code_list.py
from django.db import models

class code_list (models.Model): 
    LISTTYPE = models.CharField(null = False, max_length = 100)
    CULTURE = models.CharField(null = False, default = "Culture", max_length = 10)
    CODE = models.CharField(null = False, max_length = 100)
    CODETEXT = models.CharField(null = True, max_length = 100)
    PARENT = models.CharField(null = True, max_length = 100)
    LEGACYCODE = models.CharField(null = True, max_length = 50)
    VAL1 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL2 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL3 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL4 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL5 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL6 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL7 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL8 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL9 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    VAL10 = models.DecimalField(decimal_places = 12, null = True, max_digits = 28)
    DATE1 = models.DateField(null = True)
    DATE2 = models.DateField(null = True)
    DATE3 = models.DateField(null = True)
    DATE4 = models.DateField(null = True)
    DATE5 = models.DateField(null = True)
    CHAR1 = models.CharField(null = True, max_length = 1000)
    CHAR2 = models.CharField(null = True, max_length = 1000)
    CHAR3 = models.CharField(null = True, max_length = 1000)
    CHAR4 = models.CharField(null = True, max_length = 1000)
    CHAR5 = models.CharField(null = True, max_length = 1000)
    LAYER_NAME = models.CharField(null = False, max_length = 50)
    DESCRIPTION_ID = models.CharField(null = True, max_length = 100)
    HIDDEN = models.CharField(null = True, max_length = 5)
    LAST_UPDT_USER = models.CharField(null = False, default = "system", max_length = 100)
    LAST_UPDT_DATE = models.DateField(null = True, default = "2022-07-29")
    VERSION = models.CharField(null = False, default = "2dc6b96a1ff845ad97861d5bd7f73577", max_length = 32)
    DB_ID = models.CharField(null = True, max_length = 32)
    ROW_ID = models.CharField(null = False, default = "353298689add4c00aa881021e5633c33", max_length = 32)
    STATUS = models.CharField(null = True, max_length = 10)
    REV_GRP_ID = models.CharField(null = True, max_length = 32)
    

File: frontend/src/services/api/djangoApi/codeList.js
import { instance, config } from "../../baseApi";


export const getParentCodeList = async (CULTURE) => {
    const body = JSON.stringify({ CULTURE, LIST_TYPE: "CODE_LIST" });
    try {
        let res = await instance
            .post(
                "/code-list/details/",
                body,
                config
            )
        return res;

    } catch (err) {
        return err
    }
}


export const getParentCode = async (CULTURE, ROW_ID) => {
    const body = JSON.stringify({ ROW_ID });
    try {
        let res = await instance
            .post(
                "/code-list/deep-details/",
                body,
                config
            )
        return res



    } catch (err) {
        return err
    }
}

export const getChildCodeList = async (LIST_TYPE, CULTURE) => {
    const body = JSON.stringify({ CULTURE, LIST_TYPE });
    var mydata = { data: {} }
    try {
        let res = await instance
            .post(
                "/code-list/details/",
                body,
                config
            )

        return res;

    } catch (err) {
        console.log("asdsad");
        return "failed"
    }
}


export const deleteCodeList = async (ROW_ID) => {
    const body = JSON.stringify({ ROW_ID });
    console.log(body);
    try {
        let res = await instance
            .post(
                "/code-list/delete/",
                body,
                config
            )
        return res
    } catch (err) {
        return false
    }
}

export const putCodeList = async (CODE, CODE_TEXT, CULTURE, LIST_TYPE, ROW_ID, PARENT, LEGACY_CODE, VAL1, VAL2, VAL3, DATE1,
    DATE2, CHAR1, CHAR2, LAYER_NAME, LAST_UPDT_USER
) => {
    const body = JSON.stringify({
        LIST_TYPE, CULTURE, CODE, CODE_TEXT, ROW_ID, PARENT, LEGACY_CODE, VAL1, VAL2, VAL3,
        //DATE1, DATE2, 
        CHAR1, CHAR2,
        LAYER_NAME, LAST_UPDT_USER
    });
    console.log(body);
    try {
        let res = await instance
            .put(
                "/code-list/save/",
                body,
                config
            )
        return res
    } catch (err) {
        return false
    }
}